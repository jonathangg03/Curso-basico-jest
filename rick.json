{
    "id": 1,
    "name": "Summer Smith",
    "status": "Alive",
    "species": "Human",
    "gender": "Male"
}