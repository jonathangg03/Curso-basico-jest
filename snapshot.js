export const getCharacter = (data) => {
    const objeto = {
        "id": data.id,
        "name": data.name,
        "status" : data.status,
        "species": data.species,
        "gender": data.gender
    }
    
    return objeto
};