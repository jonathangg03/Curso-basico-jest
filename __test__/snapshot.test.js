import { getCharacter } from '../snapshot';
import rick from '../rick.json';

describe('Es hora de las instantaneas', () => {
    test('Snapshoot', () => {
        expect(getCharacter(rick)).toMatchSnapshot();
    });

    test('Siempre fallará la instantanea', () => {
        const user = {
            createAt: new Date(),
            id: Math.floor(Math.random() * 20)//Valores aleatorios del 0 al 19
        }
        expect(user).toMatchSnapshot();
    });


    test('Tenemos una excepción', () => {
        const user = {
            id: Math.floor(Math.random() * 20),//Valores aleatorios del 0 al 19
            name: 'Jonathan Garcia'
        }

        expect(user).toMatchSnapshot({
            id: expect.any(Number)
        })
    });
});