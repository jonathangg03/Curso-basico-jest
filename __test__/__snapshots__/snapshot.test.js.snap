// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Siempre fallará la instantanea 1`] = `
Object {
  "createAt": 2020-12-01T04:22:48.862Z,
  "id": 6,
}
`;

exports[`Es hora de las instantaneas Snapshoot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Summer Smith",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las instantaneas Tenemos una excepción 1`] = `
Object {
  "id": Any<Number>,
  "name": "Jonathan Garcia",
}
`;
